spec/lib/hubspot/blog_spec.rb

```diff
@@ -1,116 +1,150 @@
-      url = Hubspot::Connection.send(:generate_url, Hubspot::Blog::BLOG_LIST_PATH)
-      resp = HTTParty.get(url, format: :json)
-      resp.parsed_response["objects"].first
-    end
-  end
-  let(:created_range_params) { { created__gt: false, created__range: (Time.now..Time.now + 2.years)  } }
-
   before do
     Hubspot.configure(hapikey: "demo")
-    Timecop.freeze(Time.local(2012, 'Oct', 10))
+    Timecop.freeze(Time.utc(2012, 'Oct', 10))
   end
 
   after do
     Timecop.return
   end
 
   describe Hubspot::Blog do
-
     describe ".list" do
-      cassette "blog_list"
-      let(:blog_list) { Hubspot::Blog.list }
+      it "returns a list of blogs" do
+        VCR.use_cassette("blog_list", record: :none) do
+          result = Hubspot::Blog.list
 
-      it "should have a list of blogs" do
-        blog_list.count.should be(1)
+          assert_requested :get, hubspot_api_url("/content/api/v2/blogs?hapikey=demo")
+          expect(result).to be_kind_of(Array)
+          expect(result.first).to be_a(Hubspot::Blog)
+        end
       end
     end
 
     describe ".find_by_id" do
-      cassette "blog_list"
+      it "retrieves a blog by id" do
+        VCR.use_cassette("blog_list", record: :none) do
+          id = 351076997
+          result = Hubspot::Blog.find_by_id(id)
 
-      it "should have a list of blogs" do
-        blog = Hubspot::Blog.find_by_id(351076997)
-        blog["id"].should eq(351076997)
+          assert_requested :get, hubspot_api_url("/content/api/v2/blogs/#{id}?hapikey=demo")
+          expect(result).to be_a(Hubspot::Blog)
+        end
       end
     end
 
-    describe "#initialize" do
-      subject{ Hubspot::Blog.new(example_blog_hash) }
-      its(["name"]) { should == "API Demonstration Blog" }
-      its(["id"])   { should == 351076997 }
-    end
+    describe "#[]" do
+      it "returns the value for the given key" do
+        data = {
+          "id" => 123,
+          "name" => "Demo",
+        }
+        blog = Hubspot::Blog.new(data)
 
-    describe "#posts" do
-      cassette "one_month_blog_posts_filter_state"
-      let(:blog) { Hubspot::Blog.new(example_blog_hash) }
+        expect(blog["id"]).to eq(data["id"])
+        expect(blog["name"]).to eq(data["name"])
+      end
 
-      describe "can be filtered by state" do
+      context "when the value is unknown" do
+        it "returns nil" do
+          blog = Hubspot::Blog.new({})
 
-        it "should filter the posts to published by default" do
-          pending 'This test does not pass reliably'
-          blog.posts.length.should be(14)
+          expect(blog["nope"]).to be_nil
         end
+      end
+    end
 
-        it "should validate the state is a valid one" do
-          expect { blog.posts('invalid') }.to raise_error(Hubspot::InvalidParams)
-        end
+    describe "#posts" do
+      it "returns published blog posts created in the last 2 months" do
+        VCR.use_cassette("blog_posts/all_blog_posts", record: :none) do
```

Comment:
This PR introduces a `blog_posts` directory in `/vcr_cassettes` to group all blog post related cassettes. An alternative option could be to name these files `<endpoint>_<action/desc>`. Example: `blog_posts_all` and `blog_posts_filter`. I don't feel strongly about either direction.